contracts/vaults/YearnVaults/base/RibbonVault.sol

```diff
@@ -0,0 +1,672 @@
+        uint256 currentRound = vaultState.round;
+        uint256 receiptRound = depositReceipt.round;
+
+        uint256 unredeemedShares =
+            depositReceipt.getSharesFromReceipt(
+                currentRound,
+                roundPricePerShare[uint16(receiptRound)],
+                vaultParams.decimals
+            );
+
+        shares = isMax ? unredeemedShares : shares;
+        require(shares > 0, "!shares");
+        require(shares <= unredeemedShares, "Exceeds available");
+
+        // If we have a depositReceipt on the same round, BUT we have some unredeemed shares
+        // we debit from the unredeemedShares, but leave the amount field intact
+        // If the round has past, with no new deposits, we just zero it out for new deposits.
+        depositReceipts[msg.sender].amount = receiptRound < currentRound
+            ? 0
+            : depositReceipt.amount;
+
+        depositReceipts[msg.sender].unredeemedShares = uint128(
+            unredeemedShares.sub(shares)
+        );
+
+        emit Redeem(msg.sender, shares, receiptRound);
+
+        _transfer(address(this), msg.sender, shares);
+    }
+
+    /************************************************
+     *  VAULT OPERATIONS
+     ***********************************************/
+
+    /*
+     * @notice Helper function that helps to save gas for writing values into the roundPricePerShare map.
+     *         Writing `1` into the map makes subsequent writes warm, reducing the gas from 20k to 5k.
+     *         Having 1 initialized beforehand will not be an issue as long as we round down share calculations to 0.
+     * @param numRounds is the number of rounds to initialize in the map
+     */
+    function initRounds(uint256 numRounds) external nonReentrant {
+        require(numRounds < 52, "numRounds >= 52");
+
+        uint16 _round = vaultState.round;
+        for (uint16 i = 0; i < numRounds; i++) {
+            uint16 index = _round + i;
+            require(index >= _round, "Overflow");
+            require(roundPricePerShare[index] == 0, "Initialized"); // AVOID OVERWRITING ACTUAL VALUES
+            roundPricePerShare[index] = PLACEHOLDER_UINT;
+        }
+    }
+
+    /*
+     * @notice Helper function that performs most administrative tasks
+     * such as setting next option, minting new shares, getting vault fees, etc.
+     * @return newOption is the new option address
+     * @return lockedBalance is the new balance used to calculate next option purchase size or collateral size
+     */
+    function _rollToNextOption() internal returns (address, uint256) {
+        require(block.timestamp >= optionState.nextOptionReadyAt, "!ready");
+
+        address newOption = optionState.nextOption;
+        require(newOption != address(0), "!nextOption");
+
+        (uint256 lockedBalance, uint256 newPricePerShare, uint256 mintShares) =
+            VaultLifecycleYearn.rollover(
+                totalSupply(),
+                totalBalance(),
```

Comment:
I think we're meant to pass in the `collateralToken.balanceOf(address(this))` (yvUSDC balance) instead of `totalBalance()` (USDC balance)

This is likely why it's throwing off the calculation in `VaultLifecycleYearn.createShort`